src/Uri.php

```diff
@@ -671,10 +679,15 @@ public static function createFromComponents(array $components = []): self
      *
      * @param resource|null $context
      *
-     * @throws SyntaxError If the file does not exist or is not readable
+     * @throws FileinfoSupportMissing If ext/fileinfo is not installed
+     * @throws SyntaxError            If the file does not exist or is not readable
      */
     public static function createFromDataPath(string $path, $context = null): self
     {
+        if (!class_exists(\finfo::class)) {
```

Comment:
Question why not create a static variable so to make the check only once something along the line of what I did for the IDN support check 

```php
static $finfo_support = null;
$finfo_support = $finfo_support ?? class_exists(\finfo::class);

if (!$finfo_support) {
  throw ...
}
```

Reply:
works for me, updated.